### ISSUE TITLE
Implement Chinese Remainder Theorem over Cartesian Products of Vectors

### ISSUE THREAD

**Issue by czgdp1807** [sympy/sympy#18428]
There is some work on the above topic at https://github.com/sympy/sympy/pull/10949
It looked like a nice addition. It would be great someone can continue it. It is an easy to fix task.

**Comment by mijo2** [comment]
Hi @czgdp1807 , I would like to work on it. Will that be ok?

**Comment by czgdp1807** [comment]
Yes, sure. Please cherry pick the commits from the PR referenced above. Or use, `git fetch origin pull/10949/head:crt_cartesian` to make a new branch from that PR, merge master into it, resolve conflicts and make your additions.

**Comment by namannimmo10** [comment]
@mijo2 I'd apologize for raising a pull request. I didn't see your comment earlier. Feel free to fix this issue as what @czgdp1807 suggests to you. Also, feel free to revive my Pull request.

**Comment by mijo2** [comment]
Ok @namannimmo10 sure thing. @czgdp1807 In the ```crt_cartesian``` function mentioned in the above PR, a member of the community said that final sorting of the output is not needed. So, should I remove the sorting or take the function definition as it is?

**Comment by czgdp1807** [comment]
IMO, we shouldn't sort the output.

**Comment by mijo2** [comment]
Ok so removing the sorting then

**Comment by i-m-vivek** [comment]
@czgdp1807 Can I work on this issue?

**Comment by czgdp1807** [comment]
The work is already under progress by @mijo2

**Comment by EdisonMM** [comment]
Can two people work on the same issue?

**Comment by mijo2** [comment]
@EdisonMM This issue is mostly fixed by PR #18430 and it's just the question of whether to merge it or not. The crt remainder over Cartesian products of vectors is just in YAGNI status as of now.

**Comment by smichr** [comment]
> YAGNI status as of now

:-) maybe we need a label for this

**Comment by death-angel-141** [comment]
```
def crt_cartesian(rem, mod):

    lmod=len(mod)
    lrem=len(rem)

    if((lmod > lrem) or (lmod == 0)):
         raise ValueError("Few Arguments" )

    m = mod[0]       """initializing the variables
    R = rem[0]              so as to remove any 
     r=0                        garbage values
                              """
    for i in range(1, lmod):
        rem2 = []
        s = mod_inverse(m, mod[i])
        _m = m
        m *= mod[i]
        for elem in R:
            for k in rem[i]:
                r = elem
                r += _m * s * (k - r)
                r %= m
                rem2.append(r)
    return (rem2)

```

Clean Implementation of Chinese Remainder Theorem as it was already implemented

**Comment by mijo2** [comment]
@death-angel-141 The implementation is already done by PR #18430 and if you have any suggestions then you can review that PR and help me make some changes or make one of your own PR if you think it is necessary. Thanks.

**Comment by death-angel-141** [comment]
> 
> 
> @death-angel-141 The implementation is already done by PR #18430 and if you have any suggestions then you can review that PR and help me make some changes or make one of your own PR if you think it is necessary. Thanks.

Yeah that's why I already said,that its a clean version of the respected PR

**Comment by mijo2** [comment]
Ok, thanks for your suggestion, I will incorporate the changes that you are suggesting and see if it helps or not. But, for future suggestions, you can review my PR. Thanks for the help :)

**Comment by death-angel-141** [comment]
Welcome, and yeah I have already tested the changes that I made @mijo2

**Comment by mijo2** [comment]
@death-angel-141 I see, I am currently busy with something else right now, but I think what would really help when it comes to this issue is finding the use cases of crt_remainder_theorem's use-cases. I tried finding something about it but then got busy with some other issues and tasks. If you can find those use-cases then it would be help a lot and finally the maintainers will be able to merge the PR contributed by us.

**Comment by death-angel-141** [comment]
@mijo2 Yeah that could be a great idea

**Comment by Arpan612** [comment]
@czgdp1807 Sir, is any work left which I can do?

**Comment by Bilsteen** [comment]
Hey guys I am new to open source . Can someone tell me if this issue is still open? And if so can i start working on it?

**Comment by czgdp1807** [comment]
As far as I remember, there weren't any popular use cases for this theorem. See, https://github.com/sympy/sympy/pull/18430#issuecomment-626209330
If you can find any just let us know. Thanks.

**Comment by Ganesh910** [comment]
@czgdp1807 Sir, people in comment section says that this idea has already been implemented but this issue is still open. I am new to open source.

Can I work on the same because issue is still open or not because it has already been implemented.

**Comment by M-Haseeb01** [comment]
I will be happy to contribute if this issue is not already fixed

**Comment by oscarbenjamin** [comment]
Closing because I don't think this is needed.
